extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java

```diff
@@ -895,18 +827,11 @@ public void registerField(FieldInfo fieldInfo) {
                 reflectionRegistration.registerMethod(injectionPoint.getTarget().asMethod());
```

Comment:
We could move the registration to the `BuiltinBean.generateListBytecode()` because of `BuiltinBean.GeneratorContext.reflectionRegistration`.

Reply:
Moved the code, although I am not totally clear on why we need the registration (unconditionally) :shrug:

Reply:
I believe that it's because of the injection point metadata. We could optimize this later...